marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -1,278 +1,287 @@
-{-# LANGUAGE DataKinds          #-}
-{-# LANGUAGE DefaultSignatures  #-}
-{-# LANGUAGE DeriveAnyClass     #-}
-{-# LANGUAGE DeriveGeneric      #-}
-{-# LANGUAGE DerivingStrategies #-}
-{-# LANGUAGE FlexibleContexts   #-}
-{-# LANGUAGE MonoLocalBinds     #-}
-{-# LANGUAGE NamedFieldPuns     #-}
-{-# LANGUAGE OverloadedStrings  #-}
-{-# LANGUAGE RankNTypes         #-}
-{-# LANGUAGE RecordWildCards    #-}
-{-# LANGUAGE TemplateHaskell    #-}
+{-# LANGUAGE ConstraintKinds       #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DefaultSignatures     #-}
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DerivingStrategies    #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns        #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE RecordWildCards       #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
 {-# OPTIONS_GHC -Wno-name-shadowing #-}
 {-# OPTIONS_GHC -fno-ignore-interface-pragmas #-}
 {-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
 {-# OPTIONS_GHC -fno-specialise #-}
 
 module Language.Marlowe.Client where
+import           Control.Lens
+import           Control.Monad                         (void)
+import           Language.Marlowe.Semantics            hiding (Contract)
+import qualified Language.Marlowe.Semantics            as Marlowe
+import           Language.Plutus.Contract
+import           Language.Plutus.Contract.StateMachine (AsSMContractError, StateMachine (..), Void)
+import qualified Language.Plutus.Contract.StateMachine as SM
+import qualified Language.PlutusCore.Universe          as PLC
+import qualified Language.PlutusTx                     as PlutusTx
+import           Language.PlutusTx.AssocMap            (Map)
+import qualified Language.PlutusTx.AssocMap            as Map
+import qualified Language.PlutusTx.Prelude             as P
+import           Ledger                                (CurrencySymbol, Datum (..), Slot (..), SlotRange, TokenName,
+                                                        ValidatorCtx (..), mkValidatorScript, pubKeyHash, validatorHash,
+                                                        valueSpent)
+import           Ledger.Ada                            (adaSymbol, adaValueOf)
+import           Ledger.Constraints
+import           Ledger.Interval
+import           Ledger.Scripts                        (Validator)
+import qualified Ledger.Typed.Scripts                  as Scripts
+import           Ledger.Typed.Tx                       (TypedScriptTxOut (..))
+import qualified Ledger.Value                          as Val
+
+type MarloweInput = (SlotRange, [Input])
+
+type MarloweSchema =
+    BlockchainActions
```

Comment:
very minor, but it would be nice to have explicit import of BlockChainActions for readability. I guess it's in `Language.Plutus.Contract`